src/x_system/x_report.1
.TH X_REPORT 1 "Feb 27 2014" "GMT 4.5.13 (SVN)" "Generic Mapping Tools"
.SH NAME
\fBx_report\fP \- Report Cross-Over Error Statistics for Cruises
.SH SYNOPSIS
\fBx_report\fP [ \fIleg\fP ] [ \fB\-X\fP\fIlegbase\fP ] [ \fB\-G\fP ] [ \fB\-M\fP ] [ \fB\-T\fP ]
.SH DESCRIPTION
\fBx_report\fP reads the x_system data base file and prints out statistics for each leg. The statistics
are the number of, means, and standard deviations of COEs for both the internal and external
cases. One or more data types (GMT) may be specified.
.TP
\fIleg\fP
Report on \fIleg\fP only. [Default is all legs].
.TP
\fB\-X\fP
Specify alternate leg data base [Default is \fIxx_legs.b\fP].
.TP
\fB\-G\fP
Give information for Gravity cross-overs only. [Default is G, M, and T].
.TP
\fB\-M\fP
Give information for Magnetics cross-overs only. [Default is G, M, and T].
.TP
\fB\-T\fP
Give information for Topography cross-overs only. [Default is G, M, and T].
.SH SEE ALSO
.IR GMT (1),
.IR x_system (1)
.SH REFERENCES
Wessel, P. XOVER: A Cross-over Error Detector for Track Data,
\fIComputers & Geosciences, 15\fP, 333-346.


ChangeLog
-----------------------------------------------------------------------------------------
	$Id: ChangeLog 12925 2014-02-20 22:02:24Z pwessel $
-----------------------------------------------------------------------------------------

-----------------------------------------------------------------------------------------
201x-yy-zz	pwessel
	GMT 5.1.1 finalized and released.
-----------------------------------------------------------------------------------------
2014-02-20	pwessel
	* grdcontour.c:		The -Zscale/offset option was applied incorrectly due to a
				change in the underlying scaling function.
	* grdvolume.c:		---"---
2014-02-10	pwessel
	*gmt2kml.c:		Make sure the default style IDs are unique for each process.

2014-02-03	pwessel
	* pslib.c:		Auto-widen paragraph width to at least fit widest word.

2014-01-26	pwessel
	* gmt_grdio.c:		Building the grid command string could exceed its length if
				the argument became exactly the max length of the array; then
				a final extra character \0 would be added and exceed array length.
				Thanks to Joachim Saul for pointing it out.

2014-01-22	pwessel
	* pscoast.c:		The processing of -I -N levels and pens could get out of sync
				so that the wrong pen was used for the specified feature.
2014-01-16	remko
	* pslib.c:		Take superscripts and subscripts into account when determining
				dimensions of text box.

2014-01-10	pwessel
	* grdmath.c:		We consider the change in SDIST, PDIST to return degrees instead
				of km to be a bug as it makes it difficult to handle geodesics
				for which degrees make no sense.  Thus we have consolidated all
				the ?DIST operators to give km if geographic data are involved.
				Example 23 has been adjusted accordingly, as has several test
				scripts.  Please check your own scripts for SDIST.

2014-01-09	pwessel
	* gmt_vector.c:		Add protection against introduced longitude-jumps produced
				by resampling in GMT_fix_up_path.
2014-01-09	remko
	* gmt_vector.c:		Add protection against introduced longitude-jumps produced
				by resampling in GMT_fix_up_path.
	* grdmath.c:		LDIST and PDIST will again produce km when using geodesics.

2014-01-01	pwessel
	* x2sys/x2sys.c:	Mishandled the assignment of segment number for each record.
	* gmt_support.c:	Allow GMT_crossover to deal with global wrapped data.

2013-12-31	pwessel
	* gmt2kml.c:		Now using GMT_Put_Record instead of old stdout printing.
				This addresses issue #473.

2013-12-10	pwessel
	* grdvector.c:		Vector direction did not adjust when negative scales were
				used in -JX or -Jx.
	* psxy[z].c		---"---

2013-12-08	remko
	* psimage.c:		Now properly plots transparent images and makes -Gt work.

2013-11-26	pwessel
	* gmt_io.c:		GMT would crash on a blank line with no newline at end at all.
	* pstext.c:		pstext did not process GMT4-style input properly.

2013-11-26	jluis
	* potential/gravfft.c:	Let -E options default to 3.
				Let users select the -N+l (leave trend alone). That is, do not overwrite with default mode.
				Let the default detrend be remove MID point.
				Explicitly set -N+a in gravfft command to avoid failures due to coming changes in gravfft.
2013-11-25	pwessel
	* grdlandmask.c:	Failed to fill partial bins with no coastlines when the region
				crossed Greenwich [issue #451].

2013-11-21	pwessel
	* pscontour.c:		Let pscontour output contours to file [issue #448].
	* pscoast.c:		Prevent users from not giving proper projection [issue #449].

2013-11-19	jluis
	* mgd77/mgd77magref.c:	Needed to reset the t_col variable when constant altitude [issue #446].

2013-11-19	pwessel
	* gmt_sph.c:		Fix invalid read in ssrfpack_grid due to par array being of length 2 rather than 3.

2013-11-18	pwessel
	* gmt_grdio.c:		Place the old auto-detect of 4-byte binary float vs int under compatibility mode for backwardness.

2013-11-16	pwessel
	* grdvector.c:		Address shortcomings of geovectors when wrapped and clipped.

2013-11-15	pwessel
	* geovectors.c:		Fix geovector when shrinking is selected.

2013-11-13	pwessel
	* gmt_api.c:		Fix API issues pointed out by Ian.

2013-11-13	fwobbe
	* GMT.app:		Inherit user shell in GMT.app.

2013-11-12	pwessel
	* gmt_init.c:		grdmath could corrupt the gmt.history file [issue # 438].
 	
2013-11-10	pwessel
	* gmt_support.c:	Fix symbol-size issue in psxy -Sc [issue #435].
	
2013-11-09	pwessel
	* grdvector.c:		Did not convert direction to azimuths for geo vectors.

2013-11-08	pwessel
	* grdvector.c:		Fix grdvector -I problem [issue #434].
	* ps2raster.c:		Under Mavericks, strcpy would exit if the to and from pointers
-----------------------------------------------------------------------------------------
2013-11-05	pwessel
	GMT 5.1.0 finalized and released.
-----------------------------------------------------------------------------------------

2013-11-01	remko
	* example_10.sh:	Changed input to number of languages; multi-color columns

2013-10-31	pwessel
	* gmt_dcw.c:		Added option to specify code as =<continent> which will
				automatically be expanded to all the countries in that
				continent.  E.g., -F=AF+gred will paint Africa read.
				Fixed memory leaks.

2013-10-23	pwessel
	* gmtinfo.c:		The new name of minmax.c, the latter will be honored
				when GMT compatibility mode is 4.
	* grdmath.c:		Fixed porting bugs for CURV and D2DY2; added test ops.sh

2013-10-22	pwessel
	* minmax.c:		Added -D to adjust region found with -I
	* makecpt.c:		Better reporting of gaps or overlaps in CPTs

2013-10-05	pwessel
	* greenspline.c:	Needed -f in order to select -f0T input, plus it made assumptions
				about lon,lat despite not being selected.

2013-10-04	pwessel
	* gmt.history:		This replaces the old .gmtcommands file name.

2013-09-28	remko
	* gmt_init.c:		Moved origin of JD by one day (24 Nov instead of 25 Nov).
	* ConfigUserTemplate.cmake:	Added option GMT_DOCS_DEPEND_ON_GMT

2013-09-27	pwessel
	* share/cpt:		Added 10 new CPTs, mostly for bathymetry and topography.

2013-09-14	pwessel

src/x_system/x_report.c
/*	$Id: x_report.c 9545 2011-07-27 19:31:54Z pwessel $
 *
 * XREPORT reads the xx_legs.b file and reports statistics for one or
 * all or the legs. The information reported are legname, year-of-cruise,
 * agency, and # cross-overs, mean, and st.dev for the datatypes selected.
 * If no data types are selected, all of them are reported.
 *
 * Author:	Paul Wessel
 * Date:	10-NOV-1987
 * Last rev:	06-MAR-2000	POSIX
 *
 */
 
#include "gmt.h"
#include "x_system.h"

int main (int argc, char *argv[])
{
	GMT_LONG end = FALSE, i, all_legs = TRUE, error = FALSE;
	size_t legsize = sizeof(struct LEG);
	char leg[10], gmt[3], datatype[3];
	FILE *fp = NULL;
	struct LEG thisleg;

	gmt[0] = gmt[1] = gmt[2] = FALSE;
	datatype[0] = 'G';
	datatype[1] = 'M';
	datatype[2] = 'T';

	for (i = 1; i < argc; i++) {
		if (argv[i][0] == '-') {
			switch (argv[i][1]) {
				case 'X':
					fp = fopen(&argv[i][2], "rb");
					break;
				case 'G':
					gmt[0] = TRUE;
					break;
				case 'M':
					gmt[1] = TRUE;
					break;
				case 'T':
					gmt[2] = TRUE;
					break;
				default:
					error = TRUE;
					break;
			}
		}
		else {
			strcpy(leg, argv[i]);
			all_legs = FALSE;
		}
	}
	if (error) {
		fprintf(stderr,"xreport - Print report of crossover information\n\n");
		fprintf(stderr,"Usage : xreport [<leg>] [-X<xlegsfile>] [-G] [-M] [-T]\n");
		fprintf(stderr,"	<leg> is the desired legs id. [Default is all legs]\n");
		fprintf(stderr,"	-X to specify alternate xx_legs.b file\n");
		fprintf(stderr,"	-G -M -T to report on gravity/magnetics/topography [Default is G, M, and T]\n");
		exit (EXIT_FAILURE);
	}
	if (fp == NULL && (fp = fopen ("xx_legs.b", "rb")) == NULL) {
		fprintf (stderr, "Could not find file xx_legs.b\n");
		exit (EXIT_FAILURE);
	}
  
  	/* If none selected, the default is all 3 data types */
  
  	if (gmt[0] == FALSE && gmt[1] == FALSE && gmt[2] == FALSE) gmt[0] = gmt[1] = gmt[2] = TRUE;
  
  	printf ("leg\tyear\tagency\tdata\tnx_I\tmean_I\tstdev_I\tnx_E\tmean_E\tstdev_E\n");
	while (!end && fread((void *)&thisleg, (size_t)legsize, (size_t)1, fp) == 1) {
		if (all_legs || !strcmp(thisleg.name, leg)) {
			for (i = 0; i < 3; i++) {
				if (!gmt[i]) continue;
				printf("%s\t%d\t%s\t",
					thisleg.name,
					thisleg.year,
					thisleg.agency);
				printf("%c\t%d\t%8.2f\t%7.2f\t%6d\t%8.2f\t%7.2f\n",
					datatype[i],
					thisleg.n_gmtint[i],
					thisleg.mean_gmtint[i],
					thisleg.st_dev_gmtint[i],
					thisleg.n_gmtext[i],
					thisleg.mean_gmtext[i],
					thisleg.st_dev_gmtext[i]);
			}
			if (!all_legs) end = TRUE;
		}
	}
	fclose (fp);
	exit (EXIT_SUCCESS);
}

